code/modules/events/anomaly.dm

```diff
@@ -5,6 +5,11 @@
 
 /datum/event/anomaly/proc/findEventArea()
 	var/static/list/allowed_areas
+	var/static/list/existing_areas = list()
+	for(var/area/AR in world)
+		var/turf/picked = safepick(get_area_turfs(AR.type))
+		if(picked && is_station_level(picked.z))
+			existing_areas += AR
```

Comment:
Since it's a static list, you can do this so that it only generates once. That way it'll use that pre-made list for any future anomalies rather than having to do the whole loop again:
```suggestion
	var/static/list/existing_areas
	if(!existing_areas)
		existing_areas = list()
		for(var/area/AR in world)
			var/turf/picked = safepick(get_area_turfs(AR.type))
			if(picked && is_station_level(picked.z))
				existing_areas += AR
```
Or alternatively, if you want it to update the list of available areas between anomaly events you could just remove the `/static` part in the variable.

Reply:
Good idea, no need to generate the list multiple times.